prover/vk_setup_data_generator_server_fri/src/keystore.rs

```diff
@@ -38,17 +39,15 @@ pub enum ProverServiceDataType {
 /// - large setup keys, used during proving.
 pub struct Keystore {
     /// Directory to store all the small keys.
-    basedir: String,
+    basedir: PathBuf,
     /// Directory to store large setup keys.
     setup_data_path: Option<String>,
 }
 
-fn get_base_path_from_env() -> String {
-    let zksync_home = std::env::var("ZKSYNC_HOME").unwrap_or_else(|_| "/".into());
-    format!(
-        "{}/prover/vk_setup_data_generator_server_fri/data",
-        zksync_home
-    )
+fn get_base_path_from_env() -> PathBuf {
+    locate_workspace()
+        .unwrap_or_else(|| ".".into())
+        .join("/prover/vk_setup_data_generator_server_fri/data")
 }
```

Comment:
Like with the contract verifier, this looks unnecessarily complicated and weird; it would be better to provide the relevant path via config to the binary instead.